ensembl-track-api.openapi.yaml

```diff
@@ -43,15 +43,31 @@ paths:
             application/json:
               schema:
                 type: object
-                properties:
-                  track_categories:
-                    type: array
-                    items:
-                      $ref: '#/components/schemas/TrackCategory'
+                $ref: '#/components/schemas/TrackCategories'
         '400':
           description: Invalid request.
         '404':
           description: Specified genome ID was not found.
+      post:
+        summary: Creates new track categories for a given genome.
+        requestBody:
+          description: The track categories to create.
+          required: true
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TrackCategories'
+        responses:
+          '201':
+            description: The track categories were successfully created.
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/TrackCategories'
+          '400':
+            description: Invalid request.
+          '404':
+            description: Specified genome ID was not found.
```

Comment:
Does it mean we are going to validate the genome_uuid  if it valid/exist or not ?

Reply:
This endpoint updates tracks for genomes that already exist in Track API database. This is useful for staged updates, e.g. we first submit static tracks (webprod in the future), followed by variation tracks (submitted by us or variation team). That ofcourse, needs genomes to be created first. Forgot that endpoint, adding it shortly.

Reply:
Does it mean there would be 3 endpoints then ? 

1. Create Genome
2. Create Static Tracks for genome
3. Crate Variation track for genome

If so it is not reflected in the OpenAPI Spec ?

Reply:
- 2. and 3. are the same endpoint (`/track_categories/{genome_id}`)
- Endpoint 1. is missing from the spec. Pushing code update shortly.

Reply:
Spec updated.

Reply:
1) This covers the case when we might need to update a single track while keeping other tracks in the category (`POST /track_categories/{genome_uuid}` would create/update the whole category of tracks)
2) Because we need to know the track ID, for which the track needs to be created first (with `POST /track_categories/{genome_uuid}`). The track IDs are assigned by Track API when creating new tracks.

Reply:
> Does it mean we are going to ensure that the genome_uuid is valid/exist ? How ?
I added `/genomes` endpoint to have genomes IDs to validate against, but that was probably a bad idea.
Instead, the endpoint should only check the ID format (uuid). If the genome ID exists the, submitted tracks would be linked to the existing genome entry, otherwise a new genome entry is created for linking.

Reply:
Flow for creating tracks with `POST /track_categories`:
- we (or production pipeline) submits all static tracks (e.g. `/track_categories/{genome_id}`, JSON: `{'genomic':[track,...],'gc':[track,...],...}`)
- variation (or us) submits variation tracks (same genome ids, but JSON only includes tracks in 'variation' category)